cmd/account/mgmt/account-list.go

```diff
@@ -0,0 +1,235 @@
+package mgmt
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/organizations"
+	"github.com/aws/aws-sdk-go/service/resourcegroupstaggingapi"
+	"github.com/openshift/osdctl/pkg/printer"
+	awsprovider "github.com/openshift/osdctl/pkg/provider/aws"
+	"github.com/spf13/cobra"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+)
+
+type accountListOptions struct {
+	awsClient    awsprovider.Client
+	m            map[string][]string
+	username     string
+	payerAccount string
+	accountID    string
+
+	flags      *genericclioptions.ConfigFlags
+	printFlags *printer.PrintFlags
+	genericclioptions.IOStreams
+}
+
+func newAccountListOptions(streams genericclioptions.IOStreams, flags *genericclioptions.ConfigFlags) *accountListOptions {
+	return &accountListOptions{
+		flags:      flags,
+		printFlags: printer.NewPrintFlags(),
+		IOStreams:  streams,
+	}
+}
+
+func newCmdAccountList(streams genericclioptions.IOStreams, flags *genericclioptions.ConfigFlags) *cobra.Command {
+	ops := newAccountListOptions(streams, flags)
+	accountListCmd := &cobra.Command{
+		Use:               "list",
+		Short:             "List out accounts for username",
+		DisableAutoGenTag: true,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(ops.complete(cmd, args))
+			cmdutil.CheckErr(ops.run())
+		},
+	}
+	ops.printFlags.AddFlags(accountListCmd)
+	accountListCmd.Flags().StringVarP(&ops.username, "user", "u", "", "LDAP username")
+	accountListCmd.Flags().StringVarP(&ops.payerAccount, "payer-account", "p", "", "Payer account type")
+	accountListCmd.Flags().StringVarP(&ops.accountID, "account-id", "i", "", "Account ID")
+
+	return accountListCmd
+}
+
+func (o *accountListOptions) complete(cmd *cobra.Command, _ []string) error {
+	if o.payerAccount == "" {
+		return cmdutil.UsageErrorf(cmd, "Payer account was not provided")
+	}
+	if o.username != "" && o.accountID != "" {
+		return cmdutil.UsageErrorf(cmd, "Cannot provide both username and account ID")
+	}
+	return nil
+}
+
+func (o *accountListOptions) run() error {
+
+	var (
+		OuID string
+	)
+	// Instantiate Aws client
+	awsClient, err := awsprovider.NewAwsClient(o.payerAccount, "us-east-1", "")
+	if err != nil {
+		return err
+	}
+
+	if o.payerAccount == "osd-staging-2" {
+		OuID = "ou-rs3h-ry0hn2l9"
+	} else if o.payerAccount == "osd-staging-1" {
+		OuID = "ou-0wd6-z6tzkjek"
+	}
+
+	o.awsClient = awsClient
+	if o.accountID != "" {
+		owner, err := o.listUserName(o.accountID)
+		if err != nil {
+			return err
+		}
+		fmt.Println(owner)
+		return nil
+	}
+
+	// Initialize hashmap
+
+	o.m = make(map[string][]string)
+
+	if o.username != "" {
+
+		user := o.username
+		o.m[user], err = o.listAccountsByUser(user)
+		if err != nil {
+			return err
+		}
+		return nil
+	}
+
+	if o.username == "" && o.accountID == "" {
+
+		o.m, err = o.listAllAccounts(OuID)
+		if err != nil {
+			return err
+		}
+		return nil
```

Comment:
We need to remove this line, or add the `for key, value...` loop that's on line 115 here.